### PR TITLE
statistics: make ticker and histogram extendible

### PR DESCRIPTION
**Pull request by Connor1996**
The changes are:
- make `StatisticsImpl` as a template
- remove internal stats due to https://github.com/facebook/rocksdb/pull/4714

### CODE REVIEW

File: monitoring/statistics.h [hunk 1]
```diff
@@ -135,4 +126,8 @@ inline void SetTickerCount(Statistics* statistics, uint32_t ticker_type,
   }
 }
 
+std::shared_ptr<Statistics> CreateDBStatistics() {
```
Comment (yiwu-arbug) [inline monitoring/statistics.h]: Let's not using template, but pass ticker_max and histogram_max as optional param to `CreateDBStatistics` (adding a `DBStatisticsOptions` struct probably), and store the two enum max as member variable in `StatisticsImpl`.

File: monitoring/statistics.h [hunk 2]
```diff
@@ -30,16 +30,7 @@
 
 namespace rocksdb {
 
-enum TickersInternal : uint32_t {
```
Comment (yiwu-arbug) [inline monitoring/statistics.h]: I think we should keep these internal ticker support, even though they are not being used. However we can make `INTERNAL_TICKER_ENUM_MAX` and `INTERNAL_HISTOGRAM_ENUM_MAX` visible in public header (include/rocksdb/statistics.h) so caller of `CreateDBStatistics` will know the start of custom ticker enum.

**Comment by yiwu-arbug** [comment]
Do you plan to send to upstream?

**Comment by Connor1996** [comment]
@yiwu-arbug Yes, until I pass the build🤣

**Comment by yiwu-arbug** [comment]
I'll take another look soon and let's merge it in our branch.

File: include/rocksdb/statistics.h [hunk 3]
```diff
@@ -539,6 +539,8 @@ class Statistics {
 };
 
 // Create a concrete DBStatistics object
+template <uint32_t TICKER_MAX = TICKER_ENUM_MAX,
+          uint32_t HISTOGRAM_MAX = HISTOGRAM_ENUM_MAX>
```
Comment (yiwu-arbug) [inline include/rocksdb/statistics.h]: to fix CI, seems we need to keep this API unchanged and make another templated version?

Comment (Connor1996) [reply include/rocksdb/statistics.h]: it's unrelated, I just forget to include the `.h` for jni

**Comment by yiwu-arbug** [comment]
Any idea on how to fix CI?

File: .travis.yml [hunk 4]
```diff
@@ -113,7 +113,7 @@ script:
       ;;
     cmake-mingw)
       sudo update-alternatives --set x86_64-w64-mingw32-g++ /usr/bin/x86_64-w64-mingw32-g++-posix;
-      mkdir build && cd build && cmake -DJNI=1 .. -DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc -DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++ -DCMAKE_SYSTEM_NAME=Windows && make -j4 rocksdb rocksdbjni
+      mkdir build && cd build && cmake -DJNI=1 .. -DCMAKE_CXX_FLAGS=-fno-asynchronous-unwind-tables -DCMAKE_C_COMPILER=x86_64-w64-mingw32-gcc -DCMAKE_CXX_COMPILER=x86_64-w64-mingw32-g++ -DCMAKE_SYSTEM_NAME=Windows && make -j4 rocksdb rocksdbjni
```
Comment (yiwu-arbug) [inline .travis.yml]: move to separate PR?

Comment (Connor1996) [reply .travis.yml]: okay, let's check whether it works first
